lib/maze/hooks/client_hooks.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Maze
+  module Hooks
+    # Provides the ability for callbacks to be provided as part of running Cucumber.
+    # These are akin to Cucumber's AfterConfiguration, Before and After hooks, but are invoked in such a way that
+    # Maze Runner's hooks do not interfere with callbacks registered by clients.
+    class ClientHooks
```

Comment:
Just as I started saying to myself "I'm surprised Rubocop didn't suggest this to me", it actually wants the nested rather than compact style.  I guess it's part of wider conversation that we can have about style one day, but I'll leave it as is for now (although I have now renamed the file to match the class name).